src/pages/public_registration/action_accordions/FindRegistration.tsx

```diff
@@ -0,0 +1,122 @@
+import { RegistrationListItemContent } from '../../../components/RegistrationList';
+
+function isTitle(query: string) {
+  return !isDoi(query) && !isHandle(query);
+}
+
+function isDoi(query: string) {
+  return query.includes('https://doi.org/');
+}
+
+function isHandle(query: string) {
+  return query.includes('https://hdl.handle.net/');
+}
+
+interface FindRegistrationProps {
+  setSelectedRegistration: (registration: Registration | null) => void;
+  selectedRegistration: Registration | null;
+  filteredRegistrationIdentifier: string;
+}
+export const FindRegistration = ({
+  setSelectedRegistration,
+  selectedRegistration,
+  filteredRegistrationIdentifier,
+}: FindRegistrationProps) => {
+  const { t } = useTranslation();
+  const [searchBeforeDebounce, setSearchBeforeDebounce] = useState('');
+  const debouncedSearch = useDebounce(searchBeforeDebounce);
+
+  const fetchQuery = {
+    title: isTitle(debouncedSearch) ? debouncedSearch : null,
+    doi: isDoi(debouncedSearch) ? debouncedSearch : null,
+    query: isHandle(debouncedSearch) ? debouncedSearch : null,
+  };
+
+  const duplicateRegistrationSearch = useQuery({
+    enabled: debouncedSearch.length > 0,
+    queryKey: ['duplicateRegistration', fetchQuery],
+    queryFn: () => fetchResults(fetchQuery),
+    meta: { errorMessage: t('feedback.error.get_registrations') },
+  });
+
+  const handleSearchAutoCompleteChange = (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    setSearchBeforeDebounce(event.target.value);
+  };
+
+  const searchResults = duplicateRegistrationSearch.data?.hits ?? [];
+
+  const searchResultNotContainingToBeDeleted = searchResults.filter(
+    (possibleDuplicate) => possibleDuplicate.identifier !== filteredRegistrationIdentifier
+  );
+
+  const defaultProps = {
+    options: searchResultNotContainingToBeDeleted,
+    getOptionLabel: (option: Registration | string) => {
+      if (typeof option === 'string') {
+        return option;
+      }
+      return option.entityDescription?.mainTitle ?? '';
+    },
+  };
+  return (
+    <Box sx={{ display: 'flex', flexDirection: 'column', gap: '1rem' }}>
+      <Typography variant="h3">{t('unpublish_actions.search_for_duplicate')}</Typography>
+      <Autocomplete
+        {...defaultProps}
+        freeSolo
+        id="search-duplicate-autocomplete"
+        loading={duplicateRegistrationSearch.isLoading && debouncedSearch.length > 0}
+        value={selectedRegistration}
+        onChange={(_event, newValue, _reason) => {
+          if (typeof newValue === 'string') {
+            setSelectedRegistration(null);
+          } else {
+            setSelectedRegistration(newValue);
+          }
+        }}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            placeholder={t('unpublish_actions.search_duplicate_facets')}
+            variant="filled"
+            onChange={(event) => handleSearchAutoCompleteChange(event)}
```

Comment:
her kan vi også fjerne kode og støtte oss på riktige typer fra MUI. Når vi drar ut sånt i en egen const hvor vi selv setter type, kan man miste presisjon ved å velge en parent type, og MUI vil ikke klage på det. Om man senere bytter data _kan_ det føre til bugs/krasj som kompilatoren ikke klarer å advare mot

```suggestion
            onChange={(event) => setSearchBeforeDebounce(event.target.value)}
```